csharp/CMakeLists.txt

```diff
@@ -233,80 +192,3 @@ install(TARGETS openstudio_translators_csharp DESTINATION CSharp/openstudio/ CON
 add_custom_target(csharp_sdk ALL
   DEPENDS ${OPENSTUDIO_CSHARP_DLL} openstudio_csharp openstudio_model_csharp openstudio_translators_csharp
 )
-
-if( BUILD_NUGET_PACKAGE )
-  if(CMAKE_SIZEOF_VOID_P EQUAL 4) # 32 bit
-    message(FATAL_ERROR "Nuget package cannot be built using the 32 bit generator")
-  endif()
-
-  if (MSVC)
-    include(ExternalProject)
-
-    ExternalProject_Add(OS-32
-      SOURCE_DIR ${PROJECT_SOURCE_DIR}
-      BINARY_DIR ${PROJECT_BINARY_DIR}/OS-32
-      BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release
-      CMAKE_GENERATOR
-        "Visual Studio 16 2019"
-      CMAKE_ARGS
-        -A Win32
-        -DCMAKE_BUILD_TYPE=Release
-      CMAKE_CACHE_ARGS
-        -DBUILD_CSHARP_BINDINGS:BOOL=ON
-        -DBUILD_NUGET_PACKAGE:BOOL=OFF
-      INSTALL_COMMAND ""
-    )
```

Comment:
This is probably problematic too. We do need to be able to build a 32 bit nuget package.

Reply:
Yes. 32 bit support is indeed still a requirement.

Reply:
Hi @jmarrec and @kbenne,
The 32 bit nuget package has been built and tested successfully.

![image](https://user-images.githubusercontent.com/9031066/126071875-822738c1-aa32-4eed-af05-ebf55410767f.png)

https://github.com/MingboPeng/OpenStudio/blob/8e019a39cc0927d51e19e68ffa8d44d61b0c6401/.github/workflows/buildCSharp.yml#L94-L105

![image](https://user-images.githubusercontent.com/9031066/126071887-9c967be7-5ec8-49b8-8400-91986749858a.png)
https://github.com/MingboPeng/OpenStudio/blob/8e019a39cc0927d51e19e68ffa8d44d61b0c6401/.github/workflows/buildCSharp.yml#L131-L141

Here are test results.

![image](https://user-images.githubusercontent.com/9031066/126071964-672f5a8b-212c-44c2-acdd-ccc1dff27bfa.png)
https://github.com/MingboPeng/OpenStudio/runs/3095884349?check_suite_focus=true

Please let me know if you have any questions.

Reply:
Hey @MingboPeng I'm interested in getting this work merged, and @tijcolem and I are working on confirming that a .net core package will be suitable for all of our clients. I'm really impressed with this work, and .net core in general.

One thing about the 32 bit build, is that we need universal 32 / 64 bit builds in a single package. On the NREL develop branch this is accomplished with cmake's external project https://github.com/NREL/OpenStudio/blob/b29bc47f437cf5f5baf9e800edb49b0552838177/csharp/CMakeLists.txt#L245. We need to retain these universal packages, although the mechanism to generate them does not need to be external project, it just seemed to be the most straightforward method at the time.